MCDataProducts/inc/GenId.hh

```diff
@@ -72,6 +72,12 @@ namespace mu2e {
       _id(id)
     {}
 
+    // c'tor from a string.
+    // If the name is not recognzied it returns the value "unknown"
```

Comment:
Is there a strong reason to allow garbage value propagate downstream instead of throwing an exception as soon as a problem is detected?

Reply:
Good catch.  Just pushed the fix.  The fix is inside GenId::findByName which is used in several other places in the code base.  Those places previously propagated a value of unknown when it encountered an unrecognized input.  Now those places will throw an exception too.  I don't know of a good way to test that these still work but I looked at them all and they were not doing any checking for validity on their own.  So this is very likely an improvement everywhere.